generator/internal/language/codec.go

```diff
@@ -20,8 +20,11 @@ import "github.com/googleapis/google-cloud-rust/generator/internal/api"
 // representations. This is used to manipulate the data the is fed into
 // templates to generate clients.
 type Codec interface {
-	// TemplateDir returns the directory containing the templates.
-	TemplateDir() string
+	// TemplatesProvider returns a function that loads templates from whatever
+	// filesystem the Codec is using.
+	TemplatesProvider() func(string) (string, error)
```

Comment:
nit: by the description I can't tell what the input/output string are here. Can this be rephrased/elaborated? Even just some parameter var name may be enough. Or sometimes when dealing with higher order functions I assign them to a type so I can better document them.